app/services/category.py

```diff
@@ -0,0 +1,76 @@
+from app.crud.category import (
+    create_category,
+    read_categories,
+    read_category_by_name,
+    read_category_by_id,
+    update_category,
+    delete_category,
+)
+from app.schemas.category import CategoryCreate, CategoryUpdate, CategoryResponse
+from sqlalchemy.ext.asyncio import AsyncSession
+from fastapi import HTTPException, status
+from typing import List
+from uuid import UUID
+
+
+async def add_category(
+    db: AsyncSession, category_data: CategoryCreate
+) -> CategoryResponse:
+    existing = await read_category_by_name(db, category_data.category_name)
+    if existing:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="이미 존재하는 카테고리입니다.",
+        )
+
+    category = await create_category(db, category_data)
+    return category
+
+
+async def select_categories(db: AsyncSession) -> List[CategoryResponse]:
+    return await read_categories(db)
+
+
+async def select_category_by_name(
+    db: AsyncSession, category_name: str
+) -> CategoryResponse:
+    category = await read_category_by_name(db, category_name)
+    if not category:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND, detail="카테고리를 찾을 수 없습니다."
+        )
+    return category
+
+
+async def select_category_by_id(
+    db: AsyncSession, category_id: UUID
+) -> CategoryResponse:
+    category = await read_category_by_id(db, category_id)
+    if not category:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND, detail="카테고리를 찾을 수 없습니다."
+        )
+    return category
+
+
+async def update_category_by_id(
+    db: AsyncSession, category_id: UUID, category_data: CategoryUpdate
+) -> CategoryResponse:
+    category = await read_category_by_id(db, category_id)
+    if not category:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND, detail="카테고리를 찾을 수 없습니다."
+        )
+
+    updated_category = await update_category(db, category_id, category_data)
+    return updated_category
```

Comment:
CRUD에서 DB 예외 던지는게 생기면 이렇게 수정하시면 됩니당.

```suggestion
    try:
        category = await read_category_by_id(db, category_id)
        if not category:
            raise HTTPException(
                status_code=status.HTTP_404_NOT_FOUND, detail="카테고리를 찾을 수 없습니다."
            )

        return await updated_category = await update_category(db, category_id, category_data)
    except Exception as e:
        raise HTTPException(
            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
            detail="데이터베이스 오류가 발생했습니다.",
        ) from e    
```

Reply:
update에는 필요 없다고 판단하여 resolve 처리합니당